src/s3poller.ts

```diff
@@ -126,26 +146,45 @@ async function processS3Contents(
   bucket: string,
   contents: any[]
 ) {
-  for (const c of contents) {
-    if (!c.Key) continue
+  // First, fetch and parse all XML files to get their timestamps
+  const filesWithTimestamps = await Promise.all(
+    contents
+      .filter(c => c.Key?.toLowerCase().endsWith('.xml') && !c.Key.includes('batchcomplete'))
+      .map(async (c) => {
+        try {
+          const { Body } = await client.send(
+            new GetObjectCommand({
+              Bucket: bucket,
+              Key: c.Key,
+            })
+          )
+          const xml = await Body?.transformToString()
+          if (xml) {
+            const $ = cheerio.load(xml, { xmlMode: true })
+            const messageTimestamp = $('MessageCreatedDateTime').first().text()
+            return { key: c.Key, xml, messageTimestamp }
+          }
+        } catch (error) {
+          console.error(`Failed to fetch/parse ${c.Key}:`, error)
+        }
+        return null
+      })
+  )
 
-    const lowerKey = c.Key.toLowerCase()
-    if (!lowerKey.endsWith('.xml') || lowerKey.includes('batchcomplete')) {
-      continue
-    }
+  // Sort by messageTimestamp, handling cases where timestamp might be missing
+  const sortedFiles = filesWithTimestamps
+    .filter(Boolean)
+    .sort((a, b) => {
+      const timestampA = a!.messageTimestamp || ''
+      const timestampB = b!.messageTimestamp || ''
+      return timestampA.localeCompare(timestampB)
+    })
 
-    const xmlUrl = `s3://` + join(bucket, c.Key)
-    const { Body } = await client.send(
-      new GetObjectCommand({
-        Bucket: bucket,
-        Key: c.Key,
-      })
-    )
-    const xml = await Body?.transformToString()
-    if (xml) {
-      console.log('parsing', xmlUrl)
-      const releases = (await parseDdexXml(source, xmlUrl, xml)) || []
-    }
+  // Process in chronological order
+  for (const file of sortedFiles) {
+    const xmlUrl = `s3://` + join(bucket, file!.key)
+    console.log('parsing', xmlUrl, 'timestamp:', file!.messageTimestamp)
+    const releases = (await parseDdexXml(source, xmlUrl, file!.xml)) || []
```

Comment:
So just curious, the _parsing_ of the files changes based on the order you parse them in? Or is it just the processing of them afterwards that matters? Feels like a serial process of parsing files is slower than it could be. But if this is doing more than parsing, I could see why it matters.